src/Compilers/CSharp/Portable/Binder/Binder_InterpolatedString.cs

```diff
@@ -205,26 +203,264 @@ bool tryBindAsHandlerType([NotNullWhen(true)] out BoundInterpolatedString? resul
             }
         }
 
+        private static bool AllInterpolatedStringPartsAreStrings(ImmutableArray<BoundExpression> parts)
+            => parts.All(p => p is BoundLiteral or BoundStringInsert { Value: { Type: { SpecialType: SpecialType.System_String } }, Alignment: null, Format: null });
```

Comment:
>BoundLiteral [](http://example.com/codeflow?start=35&length=12)

nit: Are those literals necessarily string literals, as the method name "AllInterpolatedStringPartsAreStrings" suggests?

Reply:
Yes, they are.